File: codegen/results/v002/apple-foundationmodel/20260411_222913/typescript/002_word_frequency/solution.ts
import fs from 'fs';
import path from 'path';

// Function to count word frequency
function countWordFrequency(inputFile: string): Map<string, number> {
  const wordFrequency: Map<string, number> = new Map();

  // Read the file
  const content = fs.readFileSync(inputFile, 'utf8');

  // Split content into words
  const words = content.toLowerCase().split(/\W+/);

  // Count each word frequency
  words.forEach(word => {
    wordFrequency.set(word, (wordFrequency.get(word) || 0) + 1);
  });

  return wordFrequency;
}

// Main function
async function main() {
  const inputFile = 'input/text.txt';

  try {
    const wordFrequency = await countWordFrequency(inputFile);

    // Sort words first by count descending, then alphabetically ascending
    const sortedWords = wordFrequency.entries()
      .sort((a, b) => {
        if (a[1] !== b[1]) {
          return b[1] - a[1];
        }
        return a[0].localeCompare(b[0]);
      })
      .map(([word, count]) => `${word}: ${count}`);

    // Print sorted word frequencies
    sortedWords.forEach(line => process.stdout.write(line + '\n'));
  } catch (error) {
    console.error('Error:', error);
  }
}

// Run the program
main();

File: codegen/results/v002/apple-foundationmodel/20260411_222913/python/002_word_frequency/solution.py
import string
from collections import Counter

def clean_text(text):
    # Convert to lowercase
    text = text.lower()
    # Remove punctuation using string.punctuation
    text = ''.join(char for char in text if char not in string.punctuation)
    return text

def count_word_frequencies(file_path):
    # Read the file
    with open(file_path, 'r', encoding='utf-8') as file:
        text = file.read()
    
    # Clean the text
    cleaned_text = clean_text(text)
    
    # Split text into words
    words = cleaned_text.split()
    
    # Count word frequencies
    word_count = Counter(words)
    
    # Sort words first by count descending, then alphabetically ascending
    sorted_words = sorted(word_count.items(), key=lambda item: (-item[1], item[0]))
    
    # Output results
    for word, count in sorted_words:
        print(f"{word}: {count}")

# Call the function with the specified file path
count_word_frequencies('input/text.txt')